src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -276,9 +276,10 @@ public async Task SendAsync_GetWithValidHostHeader_Success(bool withPort)
         [Fact]
         public async Task SendAsync_GetWithInvalidHostHeader_ThrowsException()
         {
-            if (PlatformDetection.IsNetCore && !UseSocketsHttpHandler)
+            if (PlatformDetection.IsNetCore && (!UseSocketsHttpHandler || LoopbackServerFactory.IsHttp2))
```

Comment:
Is this temporary? If so we should add a TODO and link to the issue for HTTP2 tests, to ensure this gets cleaned up.

If not, why doesn't it apply to HTTP2? It's true that HTTP2 doesn't send the Host header, but we do inspect the Host header and send it as the "authority" pseudo-header. I'm not clear why this wouldn't apply to HTTP2 as well.

Reply:
This test depends on behavior of external server. For HTTP/1 SSL handshake fails when name does not match. For HTTP/2 I verified that we set SNI to "hostheaderthatdoesnotmatch". But in this case, server return 400 response. Maybe this is one of tests not worth of porting (or making common)